### PR TITLE
Be contest model#10

### PR DESCRIPTION
**Pull request by patelparth1993**
1. Contest Model
2. Submission Model
3. Contest Controller

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ const indexRouter = require("./routes/index");
 const pingRouter = require("./routes/ping");
 const usersRouter = require("./routes/api/users");
 const authRouter = require("./routes/api/auth");
+const constestRouter = require("./routes/api/contest");
```
Comment (perryraskin) [inline server/app.js]: Minor spelling mistake here, oops! `constestRouter` -> `contestRouter`

File: server/app.js [hunk 2]
```diff
@@ -34,6 +35,7 @@ app.use(express.static(join(__dirname, "public")));
 
 app.use("/", indexRouter);
 app.use("/ping", pingRouter);
+app.use("/api/contest", constestRouter);
```
Comment (perryraskin) [inline server/app.js]: `constestRouter` -> `contestRouter`

File: server/models/contest.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+const contestSchema = new Schema({
```
Comment (perryraskin) [inline server/models/contest.js]: My `user` model uses uppercase like `UserSchema` and filename `User.js`. For consistency, can we agree to do it that way? Super minor thing but figured I'd mention it :)

File: server/routes/api/contest.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+const contestRouter = express.Router();
+
+const auth = require("../../middleware/auth");
+const ContestModel = require("../../models/contest");
+
+// Route: GET api/contests/
+// Desc: get all active contests
+// access:  private
+
+contestRouter.get("/contests", auth, (req, res) => {
```
Comment (perryraskin) [inline server/routes/api/contest.js]: I think this should be its own api route -> `/api/contests` instead of `/api/contest/contests`

**Comment by perryraskin** [comment]
Looks good to me otherwise! Sorry for all the comments, they are very minor things 😄

File: server/app.js (hunk 1 above)
Comment (patelparth1993) [reply server/app.js]: Done, Thanks I wouldn't have caught that in this life...haha

File: server/models/contest.js (hunk 3 above)
Comment (patelparth1993) [reply server/models/contest.js]: No worries, consistent code will help us.

File: server/routes/api/contest.js (hunk 4 above)
Comment (patelparth1993) [reply server/routes/api/contest.js]: Updated, I was afraid to include separate router for contests, but as you said the URLs are cleaner this way.

**Review by iglesias294 — APPROVED** [review]

**Review by iglesias294 — COMMENTED** [review]
Correct me if i'm wrong but do we need a submissions: { type: Array, required: false} to store references to Submissions here? One contest has many submissions, but one submission has only one contest. Or will we only reference contests from the Submission schema itself?

**Comment by patelparth1993** [comment]
I believe since we will have many submissions to a contest, we are not saving submission ref in contest (some submissions would have been fine). Also, user can submit multiple designs for same contest. Combo of contest and user will be unique for submission schema. Hence, we referenced contest and user from Submission schema itself. Correct me guys if I misunderstood.

**Pull request merged**